src/base/Syntax.ml

```diff
@@ -524,6 +524,28 @@ let canonicalize_tfun t =
   let mk_new_name counter _ = "'_A" ^ Int.to_string counter in
   rename_bound_vars mk_new_name (const @@ Int.succ) t 1
 
+(* Type equivalence *)
+let type_equiv t1 t2 =
+  let t1' = canonicalize_tfun t1 in
+  let t2' = canonicalize_tfun t2 in
+  let rec equiv t1 t2 =
+    match (t1, t2) with
+    | PrimType p1, PrimType p2 when p1 = p2 -> true
+    | TypeVar v1, TypeVar v2 when v1 = v2 -> true
```

Comment:
I'd suggest to make this fail in case e.g. we decide to add location information to `TypeVar` in the future.
```suggestion
    | TypeVar v1, TypeVar v2 -> String.equal v1 v2
```